timed/projects/serializers.py

```diff
@@ -100,6 +116,54 @@ def get_root_meta(self, resource, many):
 
         return {}
 
+    def validate(self, data):
+        """Validate the role of the user.
+
+        Check if the external employee is a manager or reviewer on the corresponding
+        project or customer when he wants to create a new task.
+
+        Check if the external employee is a manager or reviewer on the task or
+        the corresponding project or customer when he wants to update the task.
+        """
+        request = self.context["request"]
+        user = request.user
+        if (
+            request.method in ["PATCH", "PUT"]
```

Comment:
Couldn't find a way without checking the request method. Open to any suggestions.